Adds the invert() method to DVector.

kjohnson0451 opened the pull request:
Satisfies #4601 
Previous PR #4604 

This effectively allows invert() to be used on the following types:
ByteArray, IntArray, RealArray, StringArray, Vector2Array, Vector3Array, ColorArray

Added local variables and vector lock, as suggested from #4604

kjohnson0451 committed:
Adds the invert() method to DVector.

This effectively allows invert() to be used on the following types:
ByteArray, IntArray, RealArray, StringArray, Vector2Array, Vector3Array, ColorArray

reduz commented:
i think get and set always lock, you can use write() to obtain a pointer
and do these operations much faster

On Fri, May 13, 2016 at 7:26 PM, kjohnson0451 <notifications@github.com>
wrote:

> Satisfies #4601 <https://github.com/godotengine/godot/issues/4601>
> Previous PR #4604 <https://github.com/godotengine/godot/pull/4604>
>
> This effectively allows invert() to be used on the following types:
> ByteArray, IntArray, RealArray, StringArray, Vector2Array, Vector3Array,
> ColorArray
>
> Added local variables and vector lock, as suggested from #4604
> <https://github.com/godotengine/godot/pull/4604>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/godotengine/godot/pull/4654
> Commit Summary
>
>    - Adds the invert() method to DVector.
>
> File Changes
>
>    - *M* core/dvector.h
>    <https://github.com/godotengine/godot/pull/4654/files#diff-0> (18)
>
> Patch Links:
>
>    - https://github.com/godotengine/godot/pull/4654.patch
>    - https://github.com/godotengine/godot/pull/4654.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/godotengine/godot/pull/4654>
>

reduz commented:
as in:

Write w = write();

do all operations on w[index]

that alone is much more efficient, when w goes out of scope, the DVector
will re-lock itself



On Fri, May 13, 2016 at 7:36 PM, Juan Linietsky <reduzio@gmail.com> wrote:

> i think get and set always lock, you can use write() to obtain a pointer
> and do these operations much faster
>
> On Fri, May 13, 2016 at 7:26 PM, kjohnson0451 <notifications@github.com>
> wrote:
>
>> Satisfies #4601 <https://github.com/godotengine/godot/issues/4601>
>> Previous PR #4604 <https://github.com/godotengine/godot/pull/4604>
>>
>> This effectively allows invert() to be used on the following types:
>> ByteArray, IntArray, RealArray, StringArray, Vector2Array, Vector3Array,
>> ColorArray
>>
>> Added local variables and vector lock, as suggested from #4604
>> <https://github.com/godotengine/godot/pull/4604>
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>   https://github.com/godotengine/godot/pull/4654
>> Commit Summary
>>
>>    - Adds the invert() method to DVector.
>>
>> File Changes
>>
>>    - *M* core/dvector.h
>>    <https://github.com/godotengine/godot/pull/4654/files#diff-0> (18)
>>
>> Patch Links:
>>
>>    - https://github.com/godotengine/godot/pull/4654.patch
>>    - https://github.com/godotengine/godot/pull/4654.diff
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/godotengine/godot/pull/4654>
>>
>
>

reduz commented:
(no need to use lock/unlock)

On Fri, May 13, 2016 at 7:37 PM, Juan Linietsky <reduzio@gmail.com> wrote:

> as in:
>
> Write w = write();
>
> do all operations on w[index]
>
> that alone is much more efficient, when w goes out of scope, the DVector
> will re-lock itself
>
>
>
> On Fri, May 13, 2016 at 7:36 PM, Juan Linietsky <reduzio@gmail.com> wrote:
>
>> i think get and set always lock, you can use write() to obtain a pointer
>> and do these operations much faster
>>
>> On Fri, May 13, 2016 at 7:26 PM, kjohnson0451 <notifications@github.com>
>> wrote:
>>
>>> Satisfies #4601 <https://github.com/godotengine/godot/issues/4601>
>>> Previous PR #4604 <https://github.com/godotengine/godot/pull/4604>
>>>
>>> This effectively allows invert() to be used on the following types:
>>> ByteArray, IntArray, RealArray, StringArray, Vector2Array, Vector3Array,
>>> ColorArray
>>>
>>> Added local variables and vector lock, as suggested from #4604
>>> <https://github.com/godotengine/godot/pull/4604>
>>> ------------------------------
>>> You can view, comment on, or merge this pull request online at:
>>>
>>>   https://github.com/godotengine/godot/pull/4654
>>> Commit Summary
>>>
>>>    - Adds the invert() method to DVector.
>>>
>>> File Changes
>>>
>>>    - *M* core/dvector.h
>>>    <https://github.com/godotengine/godot/pull/4654/files#diff-0> (18)
>>>
>>> Patch Links:
>>>
>>>    - https://github.com/godotengine/godot/pull/4654.patch
>>>    - https://github.com/godotengine/godot/pull/4654.diff
>>>
>>> —
>>> You are receiving this because you are subscribed to this thread.
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/godotengine/godot/pull/4654>
>>>
>>
>>
>

kjohnson0451 committed:
Access items with write()
Removed set() and get() calls

reduz committed:
New reworked AnimatedSprite!

-New SpriteFrames editor, with support for drag&drop, multiple animation
sets, animation speed and loop.
-New AnimatedSprite, with support for all the new features!
AnimatedSprite3D has not been updated yet.
-Added support for drag&drop to other editors, such as resourcepreload,
sample library,  etc.

CowThing committed:
Filling in documentation.

AABB, Color, Rect2, Transform, Vector2, and Vector3.

reduz committed:
Changes to FileSystem Dock

-Replaced buttons for file actions fo RMB menu
-Added a split mode, if the dock is not sharing vertical space with
anything else
-imroved drag and drop support

reduz committed:
Experimental Right Mouse Button menu for SceneTree

Will eventually replace the buttons of that dock, so please test it!

kjohnson0451 committed:
Merge remote-tracking branch 'origin/master' into dvector_invert

Conflicts:
	tools/editor/scene_tree_dock.cpp
	tools/editor/scene_tree_dock.h
	tools/editor/scene_tree_editor.cpp

kjohnson0451 committed:
Resolving conflict with scene/gui/tree.cpp

kjohnson0451 committed:
Revert "Resolving conflict with scene/gui/tree.cpp"

This reverts commit fc85de4dbcad11460565c2de387cfe0db314d38d.

kjohnson0451 committed:
Revert "Merge remote-tracking branch 'origin/master' into dvector_invert"

This reverts commit 9625ed17c5bd05d1863764825989f87a5f45d0fa, reversing
changes made to 8ab6592cedf24cb0a337ed8024d90ac6b45099e9.

kjohnson0451 committed:
Revert "Experimental Right Mouse Button menu for SceneTree"

This reverts commit 7444a7a2e70a510a4c14eb43f97cf82cb85c0a3e.

kjohnson0451 committed:
Revert "Changes to FileSystem Dock"

This reverts commit c3ca02da41db1d910c15c068d9a314a11e8eb917.

kjohnson0451 committed:
Revert "Filling in documentation."

This reverts commit 8ab6592cedf24cb0a337ed8024d90ac6b45099e9.

kjohnson0451 committed:
Revert "New reworked AnimatedSprite!"

This reverts commit e0546f7a5f7e117e2b859ec75083d8f989cded8f.
